src/Admin/AbstractAdmin.php

```diff
@@ -3262,24 +3266,43 @@ public function canAccessObject($action, $object)
     }
 
     /**
-     * @return string[]
+     * @return MutableTemplateRegistryInterface
      */
-    protected function configureExportFields(): array
+    final public function getTemplateRegistry()
     {
-        return $this->getModelManager()->getExportFields($this->getClass());
+        // NEXT_MAJOR: Remove the deprecation and uncomment the exception.
+        if (!$this->hasTemplateRegistry()) {
+            @trigger_error(sprintf(
+                'Calling %s() when there is no templateRegistry is deprecated since sonata-project/admin-bundle 3.x'
+                .' and will throw an exception in 4.0. '
```

Comment:
```suggestion
                'Calling %s() when there is no template registry is deprecated since sonata-project/admin-bundle 3.x'
                .' and will throw an exception in 4.0.'
```